qualtran/_infra/controlled.py

```diff
@@ -396,7 +395,10 @@ def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
                 f"Could not build call graph for {self}: {e2}"
             ) from e2
 
-        return {(bloq.controlled(self.ctrl_spec), n) for bloq, n in sub_cg}
+        if isinstance(sub_cg, dict):
+            return {(bloq.controlled(self.ctrl_spec), n) for bloq, n in sub_cg.items()}
+        else:
+            return {(bloq.controlled(self.ctrl_spec), n) for bloq, n in sub_cg}
```

Comment:
maybe err on the side of returning the new-style dict rather than casting everything to old-style set

Reply:
Done.